cmd/clusterctl/client/cluster/mover.go

```diff
@@ -79,7 +79,11 @@ func (o *objectMover) Move(namespace string, toCluster Client) error {
 	if err := o.checkProvisioningCompleted(objectGraph); err != nil {
 		return err
 	}
-	//TODO: consider if to add additional preflight checks ensuring the object graph is complete (no virtual nodes left)
+
+	// Check whether they are virtual node existing
+	if err := objectGraph.checkVirtualNode(); err != nil {
+		return err
```

Comment:
I'm not really sure we should error and block in this case, what about logging only?

The main reason to log only is that this check should fail only if there are owner references pointing to objects which are not included in the list considered by CAPI, which is a violation of the clusterctl contract. So, given that, I prefer the user to be informed, but not blocked by clusterctl.

Reply:
@wfernandes opinions? ^^

Reply:
ok, make sense to me ,will update